### ISSUE TITLE
[Project 2] Question on priorities of writers

### ISSUE THREAD

**Issue by HunHuh** [swsnu/osspr2021#51]
Hi, 
I was thinking of a situation and couldn't figure out which would be the right action.
The situation is as below.

R[0, 20] acquired
R[30, 60] acquired
in this state, W[0, 60] and W[0,20] calls the lock in this order.

Then the overall state of the locks would be
R[0, 20] acquired
R[30, 60] acquired
W[0, 60] waiting
W[0, 20] waiting

In this state, if R[0, 20] is released, should the following state be
R[30, 60] acquired
W[0, 60] waiting
W[0, 20] waiting
or
R[30, 60] acquired
W[0, 60] waiting
W[0, 20] acquired
, or is either one fine?
In other words, is there any priority for the writer lock that has been called earlier?

**Comment by jaywonchung** [comment]
That's a great question. I'm glad you asked.

So the fairness policy we gave you does not answer your question. This means that you are free to do as you wish, as long as you adhere to the policy we gave you.

**Comment by HunHuh** [comment]
I see. Thank you!

**Comment by ChanwooLucioLee** [comment]
I had the same question. Thanks for asking @HunHuh.
